### PR TITLE
Generated files for sisdk 2024 06 0

### PR DESCRIPTION
**Pull request by jmartinez-silabs**
#### Problem / Feature
Updating to sisdk 2024.06.0 requires new gen files

#### Change overview
Remove MG12 gen files
regen for our support MG24,MGM24 boards
Add MG26 gen files

#### Testing
*********.  DO NOT MERGE YET ************

### CODE REVIEW

File: matter/efr32/efr32mg24/BRD2601B/autogen/sl_event_handler.h [hunk 1]
```diff
@@ -7,5 +7,6 @@ void sl_driver_init(void);
 void sl_service_init(void);
 void sl_stack_init(void);
 void sl_internal_app_init(void);
+void sl_iostream_init_instances(void);
```
Comment (mkardous-silabs) [inline matter/efr32/efr32mg24/BRD2601B/autogen/sl_event_handler.h:10]: This causes a duplicate function definition that we forgot to fix in the gsdk.
Signature is also in `autogen/sl_iostream_init_instances.h`.

File: matter/efr32/efr32mg24/BRD2601B/config/sl_power_manager_config.h [hunk 2]
```diff
@@ -57,6 +57,20 @@
...
+// <EMU_EM4CTRL_EM4IORETMODE_DISABLE=> No retention
+// <EMU_EM4CTRL_EM4IORETMODE_EM4EXIT=> Retention through EM4
+// <EMU_EM4CTRL_EM4IORETMODE_SWUNLATCH=> Retention through EM4 and wakeup
+// <d> power_manager_pin_retention_disable
+#define SL_POWER_MANAGER_INIT_EMU_EM4_PIN_RETENTION_MODE EMU_EM4CTRL_EM4IORETMODE_DISABLE
+
+// <e SL_POWER_MANAGER_INIT_EMU_EM2_DEBUG_ENABLE > Enable EM2 debugging feature
+// <i> Enable or disable debugging features.
+// <i> Default: 1
+#define SL_POWER_MANAGER_INIT_EMU_EM2_DEBUG_ENABLE 1
```
Comment (mkardous-silabs) [inline matter/efr32/efr32mg24/BRD2601B/config/sl_power_manager_config.h:71]: Do we want to keep this debug define?

File: matter/efr32/efr32mg24/BRD2703A/autogen/sl_event_handler.c [hunk 3]
```diff
@@ -61,7 +53,7 @@ void sl_kernel_start(void)
 void sl_driver_init(void)
 {
   GPIOINT_Init();
-  sl_simple_button_init_instances();
+    sl_simple_button_init_instances();
```
Comment (mkardous-silabs) [inline matter/efr32/efr32mg24/BRD2703A/autogen/sl_event_handler.c:56]: weird space here

File: matter/efr32/efr32mg24/BRD4186A/autogen/sl_component_catalog.h [hunk 4]
```diff
@@ -11,31 +10,31 @@
...
-#define SL_CATALOG_DEVICE_INIT_CLOCKS_PRESENT
 #define SL_CATALOG_DEVICE_INIT_CORE_PRESENT
-#define SL_CATALOG_DEVICE_INIT_DCDC_PRESENT
 #define SL_CATALOG_DEVICE_INIT_DPLL_PRESENT
-#define SL_CATALOG_DEVICE_INIT_EMU_PRESENT
 #define SL_CATALOG_DEVICE_INIT_HFXO_PRESENT
-#define SL_CATALOG_DEVICE_INIT_LFRCO_PRESENT
-#define SL_CATALOG_DEVICE_INIT_LFXO_PRESENT
-#define SL_CATALOG_DEVICE_INIT_NVIC_PRESENT
+
```
Comment (mkardous-silabs) [inline matter/efr32/efr32mg24/BRD4186A/autogen/sl_component_catalog.h:22]:
```suggestion
```

**Review by mkardous-silabs — APPROVED** [review]

File: matter/efr32/efr32mg24/BRD2601B/config/sl_power_manager_config.h (hunk 2 above)
Comment (jmartinez-silabs) [reply matter/efr32/efr32mg24/BRD2601B/config/sl_power_manager_config.h:71]: I don't know what it enables. I'll search a bit. But probably we can set to 0

Comment (jmartinez-silabs) [reply matter/efr32/efr32mg24/BRD2601B/config/sl_power_manager_config.h:71]: It only sets a register bit to 1 when enabled. I think we can let it set if it allows us to debug em2 by default

File: matter/si91x/siwx917/BRD4338A/autogen/sl_component_catalog.h [hunk 5]
```diff
@@ -3,26 +3,28 @@
...
+#define SL_CATALOG_SIMPLE_LED_LED1_PRESENT
+#define SL_CATALOG_ULP_TIMER_PRESENT
+#define SL_CATALOG_ULP_TIMER_TIMER0_PRESENT
 #define SL_CATALOG_CMSIS_OS_COMMON_PRESENT
 #define SL_CATALOG_DEVICE_INIT_NVIC_PRESENT
 #define SL_CATALOG_EMLIB_CORE_PRESENT
 #define SL_CATALOG_EMLIB_CORE_DEBUG_CONFIG_PRESENT
 #define SL_CATALOG_FREERTOS_KERNEL_PRESENT
 #define SL_CATALOG_KERNEL_PRESENT
+#define SL_CATALOG_GLIB_PRESENT
```
Comment (mkardous-silabs) [inline matter/si91x/siwx917/BRD4338A/autogen/sl_component_catalog.h:23]: Do we know if this is necessary if the LCD is disabled?

Comment (jmartinez-silabs) [reply matter/si91x/siwx917/BRD4338A/autogen/sl_component_catalog.h:23]: probably not

Comment (jmartinez-silabs) [reply matter/si91x/siwx917/BRD4338A/autogen/sl_component_catalog.h:23]: actually this one is duplicated. there was one in the #ifdef DISPLAY_ENABLED that i didn't spot. Removed it

**Pull request merged**